### PR TITLE
Async functionality

### PR DESCRIPTION
**Pull request by gourab-aiplanet**

### CODE REVIEW

File: src/openagi/actions/tools/pubmed_tool.py [hunk 1]
```diff
@@ -69,4 +69,7 @@ def execute(self) -> str:
 			return formatted_results
 
 		except Exception as e:
-			return f"Error searching PubMed: {str(e)}"
\ No newline at end of file
+			return f"Error searching PubMed: {str(e)}"
+
+	async def async_execute(self):
+        return self.execute()
```
Comment (mukesh-aiplanet) [inline src/openagi/actions/tools/pubmed_tool.py]: formatting

File: src/openagi/actions/tools/yahoo_finance.py [hunk 2]
```diff
@@ -72,4 +72,7 @@ def execute(self) -> str:
 				return f"Invalid info_type: {self.info_type}. Supported types are: summary, price, history, info"
 				
 		except Exception as e:
-			return f"Error fetching data for {self.symbol}: {str(e)}"
\ No newline at end of file
+			return f"Error fetching data for {self.symbol}: {str(e)}"
+
+	async def async_execute(self):
+		return self.execute()
```
Comment (mukesh-aiplanet) [inline src/openagi/actions/tools/yahoo_finance.py:78]: formatting

**Review by mukesh-aiplanet — APPROVED** [review]
There a few formatting issues in 3-4 files other than that I am not reviewing the actual code because we'll just merge and test once, I will check later.

File: src/openagi/actions/tools/pubmed_tool.py (hunk 1 above)
Comment (gourab-aiplanet) [reply src/openagi/actions/tools/pubmed_tool.py]: resolved

File: src/openagi/actions/tools/yahoo_finance.py (hunk 2 above)
Comment (gourab-aiplanet) [reply src/openagi/actions/tools/yahoo_finance.py:78]: Checked in vs code and github.dev but shows correct indentation

**Pull request merged**